src/OpenTelemetry/Metrics/MetricStreamConfiguration.cs

```diff
@@ -91,11 +95,39 @@ public class MetricStreamConfiguration
         }
     }
 
+    /// <summary>
+    /// Gets or sets a positive integer value
+    /// defining the maximum number of data points allowed to
+    /// per view.
+    /// </summary>
+    /// <remarks>
+    /// Note: If there is no matching view, but the MetricReader
+    /// defines a default cardinality limit value based on the
+    /// instrument an aggregation is created for, that value
+    /// will be used. The default value of 2000 will be used
+    /// if neither the view nor the MetricReader configures
+    /// MaxMetricPointsPerMetricStream.
+    /// </remarks>
+#if EXPOSE_EXPERIMENTAL_FEATURES
+    public
+#else
+    internal
+#endif
+    int? MaxMetricPointsPerMetricStream
+    {
+        get => this.maxMetricPointsPerMetricStream;
+        set
+        {
+            if (value != null)
+            {
+                Guard.ThrowIfOutOfRange(value.Value, min: 1, max: int.MaxValue);
```

Comment:
i don't fully recollect Metrics design :( - but I think we do reserve 1 or 2 points for the zero tag and the overflow. should we ensure the minimum is 3 here to account for that?

Reply:
Good topic for another PR 👍

Reply:
Will use a follow-up PR to address this. 
Related discussion: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5312#discussion_r1478861696